jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunExecutorActor.scala

```diff
@@ -46,36 +42,25 @@ class JobRunExecutorActor(
   val runSpecId = jobRun.id.toPathId
 
   override def preStart(): Unit = {
-    val startingDeadline = computeStartingDeadline
-    if (startingDeadline.exists(d => d.toSeconds <= 0)) {
-      log.info(s"StartingDeadline timeout of ${jobRun.startingDeadline.get} expired for JobRun ${jobRun.id} created at ${jobRun.createdAt}")
-      becomeAborting()
-    } else {
-      startingDeadline.foreach(scheduleStartingDeadline)
+    // only 2 ways we should be here... 1) newly launched jobrun and 2) startup while last known jobruns were launched
+    log.info(s"Starting jobrun ${jobRun} with id ${jobRun.id} status: ${jobRun.status}")
+
+    jobRun.status match {
+      // on new jobrun
+      case JobRunStatus.Initial => becomeCreating()
+
+      // previously launched job (from zk) only possible with restart
+      case JobRunStatus.Starting =>
+        println(s"Initializing Actor for existing jobrun ${jobRun} in Starting state not in the queue.")
+        context.become(creating)
```

Comment:
I don't think this would work... [ersistence actor will never send the expected messages so we would never move out of this state

Reply:
maybe you meant starting...

Reply:
nope...   The secret here and the reason it works is this is `preStart` which initializes this actor.     This is called for 2 reasons that I can think of :)
1. We are starting a jobrun (which has never been started before).  It will come into `preStart` with the Initial and become creating
2. we are restarting metronome where jobruns were previously started.   This actor is initialized with a jobrun in an undefined status (it is whatever the status is at the time metronome died.   In the case of Initial everything is as normal.   In the case of Starting,  things are different.   1 of 2 things are true... the task was never launched in which case we need to create it again... or the task was launched but we died before the status update to active occurred.   Until I wrote this response I had not considered the latter case but was coded for the former case.

Once we leave `preStart`.    lifecycle is managed by receive.
now I need to think about the other part of Starting case...